src/Storages/RocksDB/StorageEmbeddedRocksDB.h

```diff
@@ -83,6 +83,10 @@ class StorageEmbeddedRocksDB final : public IStorage, public IKeyValueEntity, Wi
 
     bool supportsDelete() const override { return true; }
 
+    bool supportsTrivialCountOptimization() const override { return true; }
```

Comment:
Shouldn't we return true/false depending on a value of setting?

Reply:
It's a query settings, so we will need to pass `Settings` object as argument of  `supportsTrivialCountOptimization`. I don't think it worth doing so. Wdyt?

Reply:
Indeed, passing settings there may be not the best solution. So, it will work anyway because `totalRows` returns optional and if it's nullopt, we will execute `count()` with a fullscan anyway, right?

Reply:
Yes. That's the logic I implemented in this PR.